### PR TITLE
Leads management

### PR DESCRIPTION
**Pull request by snebo**
@CodiumAI-Agent /describe

### CODE REVIEW

File: src/lead_registration/dto/create-lead_registration.dto.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  role: string;
+
+  @ApiProperty({
+    description: 'status of registration', 
+    example: 'pending',
+    enum: ['pending', 'approved', 'rejected']
+  })
+  @IsString()
+  @IsNotEmpty()
+  @IsIn(['pending', 'approved', 'rejected'])
```
Comment (willypelz) [inline src/lead_registration/dto/create-lead_registration.dto.ts:20]: @snebo we need to fine-tune, the linking of the leads to users, or will the leads be created to their own entity.

Comment (Roxie-32) [reply src/lead_registration/dto/create-lead_registration.dto.ts:20]: @snebo what questions are we referring to here?

**Comment by snebo** [comment]
I modified the whole lead registration process to account for both scenarios(new user and existing user). it uses a Temp schema to hold the registrations until they are approved by and admin.

**Comment by snebo** [comment]
@CodiumAI-Agent /describe

**Comment by snebo** [comment]
> @CodiumAI-Agent /describe

why dosen't this work😫. It's suppoed to neatly format and list out all the changes that were made

File: src/lead_registration/dto/create-lead_registration.dto.ts (hunk 1 above)
Comment (Tonykaynoni) [reply src/lead_registration/dto/create-lead_registration.dto.ts:14]: The user role is already doing that

Comment (Tonykaynoni) [reply src/lead_registration/dto/create-lead_registration.dto.ts:14]: And le'ts follow the same naming convention fro this file, update to "create-lead-registration.dto.ts"

File: src/shared/schema/temp_lead_registration.schema.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  role:string = UserRole.LEAD;
+
+  @Prop ({required: true})
+  CreatedAt: Date;
+
+  @Prop ({required: true})
+  userId: string;
+
+  @Prop ({required: true})
+  leadPosition: string;
```
Comment (Tonykaynoni) [inline src/shared/schema/temp_lead_registration.schema.ts:20]: I'm asumming temp-lead is for those that are requesting to join the community as a lead,
Include their first and last name here, and all other details, 

1. Firstname 
2. Lastname 
3. Email.
4. whyJoinUs
5. Location.
6. Student or Porfessional 
7. Links to online profiles.
8. Cv file upload

I think this details sholud be enough for anyone planning to join us.

Comment (snebo) [reply src/shared/schema/temp_lead_registration.schema.ts:20]: Temp-lead holds tempory user(leads) applications.
the pending applications that an admin can view are all stored here.
When an applicationis approved, a new lead is created  and the bulk of its data is gotten form the users or the provided form -for new users-

**Review by willypelz — APPROVED** [review]

**Comment by Tonykaynoni** [comment]
We need to revert this PR, myself and kenny saw some errors here

**Pull request merged**